src/pint/toa.py

```diff
@@ -1728,11 +1728,12 @@ def write_TOA_file(self, filename, name="unk", format="tempo2"):
             obs_obj = Observatory.get(obs)
 
             flags = flags.copy()
-            if "clkcorr" in flags.keys():
+            if "clkcorr" in flags:
                 toatime_out = toatime - time.TimeDelta(flags["clkcorr"])
             else:
                 toatime_out = toatime
-            out_str = format_toa_line(
+            out_str = "C " if ("ignore" in flags or "cut" in flags) else ""
```

Comment:
Can you make the "ignore" flag an optional argument to `write_TOA_flag`? So that users don't find their TOAs disappearing unexpectedly, and so that users *can* write files that include `-cut` TOAs that aren't commented out?

Reply:
How will people find their TOAs disappearing unexpectedly given that 1) they have to explicitly add these flags (or they would have had to do something non-standard to the .tim file (i.e. uncommenting them) before reading them in) and 2) they don't disappear, they are just commented?  The whole point of this change is to make a simple way that we can select against TOAs that we don't want to fit or whatever, and which we can easily flag to be commented *if* we re-write the TOAs.

I'm not seeing an advantage to an optional argument (which users have to remember) -- especially since if "-cut" means that we've cut something, we really don't want it included!  If you want to make a note about a TOA, use a different flag.

Reply:
If we want to make an *additional* way of commenting TOAs via an optional argument which provides a mask array, I suppose that is fine.  The utility and ability to use flags for this type of thing is exactly why TOAs have flags now.

Reply:
My thought was that if a user has a tim file in which some TOAs have a `-cut` flag that they did not add, they may be surprised that those TOAs are commented out when the TOAs are written out, and they will have disappeared when the user tries to read the tim file back in. 

I also anticipate wanting to write out a tim file with `-cut` flags indicating reasons for cutting TOAs but with those TOAs not actually commented out, so that this tim file can easily be used to evaluate our excision criteria. So I'd like to be able to turn off the commenting feature.

Reply:
This was my point at the beginning of the discussion in #1011.  We are trying to use comments and/or the `-cut` flag to do things which they (IMO) shouldn't be doing.  NANOGrav started using the `-cut` flag to describe *why* we cut (thereby commenting out) a TOA.  But the bottom line is that `-cut` == commented as defined by NANOGrav.

So for your first point, the user shouldn't have TOAs that have a cut flag and are un-commented unless they (or someone else) modified them.   But even if they do, they still need to do the non-standard thing of writing out a new TOA file *and* be ignorant as to what the flags contain and mean in order to be "surprised".

If you (or someone else) wants to evaluate *possible* reasons to cut TOAs and test them (i.e. your 2nd point), then you should be using a different flag since all strings are valid.  But `-cut` means that we have cut the TOA!  Please don't change  the definition of `-cut`.